test/rabbitmq_queues_cli_SUITE.erl

```diff
@@ -0,0 +1,156 @@
+
+-compile(export_all).
+
+all() ->
+    [
+        {group, tests}
+    ].
+
+groups() ->
+    [
+        {tests, [], [
+            shrink
+        ]}
+    ].
+
+init_per_suite(Config) ->
+    rabbit_ct_helpers:log_environment(),
+    rabbit_ct_helpers:run_setup_steps(Config).
+
+end_per_suite(Config) ->
+    rabbit_ct_helpers:run_teardown_steps(Config).
+
+init_per_group(tests, Config0) ->
+    NumNodes = 3,
+    Config1 = rabbit_ct_helpers:set_config(
+                Config0, [{rmq_nodes_count, NumNodes},
+                          {rmq_nodes_clustered, true}]),
+    rabbit_ct_helpers:run_steps(Config1,
+                                rabbit_ct_broker_helpers:setup_steps() ++
+                                rabbit_ct_client_helpers:setup_steps()
+    ).
+
+end_per_group(tests, Config) ->
+    rabbit_ct_helpers:run_steps(Config,
+                                rabbit_ct_client_helpers:teardown_steps() ++
+                                    rabbit_ct_broker_helpers:teardown_steps()).
+
+init_per_testcase(Testcase, Config0) ->
+    ensure_rabbitmq_queues_cmd(
+      rabbit_ct_helpers:testcase_started(Config0, Testcase)).
+
+end_per_testcase(Testcase, Config0) ->
+    rabbit_ct_helpers:testcase_finished(Config0, Testcase).
+
+shrink(Config) ->
+    NodeConfig = rabbit_ct_broker_helpers:get_node_config(Config, 2),
+    Nodename2 = ?config(nodename, NodeConfig),
+    Ch = rabbit_ct_client_helpers:open_channel(Config, Nodename2),
+    %% declare a quorum queue
+    QName = "shrink1",
+    #'queue.declare_ok'{} = declare_qq(Ch, QName),
+    {ok, Out1} = rabbitmq_queues(Config, 0, ["shrink", Nodename2]),
+    ?assertMatch(#{{"/", "shrink1"} := {2, ok}}, parse_result(Out1)),
+    Nodename1 = rabbit_ct_broker_helpers:get_node_config(Config, 1, nodename),
+    {ok, Out2} = rabbitmq_queues(Config, 0, ["shrink", Nodename1]),
+    ?assertMatch(#{{"/", "shrink1"} := {1, ok}}, parse_result(Out2)),
+    Nodename0 = rabbit_ct_broker_helpers:get_node_config(Config, 0, nodename),
+    {ok, Out3} = rabbitmq_queues(Config, 0, ["shrink", Nodename0]),
+    ?assertMatch(#{{"/", "shrink1"} := {1, error}}, parse_result(Out3)),
+    ok.
+
+parse_result(S) ->
+    %% strip preamble and header
+    [_, _ |Lines] = string:split(S, "\n", all),
+    maps:from_list(
+      [{{Vhost, QName},
+        {erlang:list_to_integer(Size), case Result of
+                                           "ok" -> ok;
+                                           _ -> error
+                                       end}}
+       || [Vhost, QName, Size, Result] <-
+          [string:split(L, "\t", all) || L <- Lines]]).
+
+declare_qq(Ch, Q) ->
+    Args = [{<<"x-queue-type">>, longstr, <<"quorum">>}],
+    amqp_channel:call(Ch, #'queue.declare'{queue = list_to_binary(Q),
+                                           durable = true,
+                                           auto_delete = false,
+                                           arguments = Args}).
+ensure_rabbitmq_queues_cmd(Config) ->
```

Comment:
This feels like it could be moved to `rabbitmq_ct_helpers` and thus easy to reuse.